File: bindings/python/testsuite/sid.py
#!/usr/bin/python

import unittest
from storage import Devicegraph, Disk, GPT, DeviceNotFound, HolderNotFound


class TestSid(unittest.TestCase):

    def test_sid(self):

        devicegraph = Devicegraph()
        sda = Disk.create(devicegraph, "/dev/sda")
        gpt = sda.create_partition_table(GPT)

        self.assertEqual(sda.get_sid(), 42)
        self.assertEqual(gpt.get_sid(), 43)

        self.assertEqual(devicegraph.find_device(42).get_sid(), 42)

        self.assertRaises(DeviceNotFound, lambda: devicegraph.find_device(99))

        self.assertEqual(devicegraph.find_holder(42, 43).get_source_sid(), 42)
        self.assertEqual(devicegraph.find_holder(42, 43).get_target_sid(), 43)

        self.assertRaises(HolderNotFound, lambda: devicegraph.find_holder(99, 99))


if __name__ == '__main__':
    unittest.main()

File: bindings/ruby/testsuite/sid.rb
#!/usr/bin/ruby

require 'test/unit'
require 'storage'


class TestSid < Test::Unit::TestCase

  def test_sid

    environment = Storage::Environment.new(true, Storage::ProbeMode_NONE, Storage::TargetMode_DIRECT)
    storage = Storage::Storage.new(environment)

    devicegraph = Storage::Devicegraph.new(storage)
    sda = Storage::Disk::create(devicegraph, "/dev/sda")
    gpt = sda.create_partition_table(Storage::PtType_GPT)

    assert_equal(sda.sid, 42)
    assert_equal(gpt.sid, 43)

    assert_equal(devicegraph.find_device(42).sid, 42)

    assert_raises(Storage::DeviceNotFoundBySid) { devicegraph.find_device(99) }

    assert_equal(devicegraph.find_holder(42, 43).source_sid, 42)
    assert_equal(devicegraph.find_holder(42, 43).target_sid, 43)

    assert_raises(Storage::HolderNotFoundBySids) { devicegraph.find_holder(99, 99) }

  end

end
